network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/pixels/WireguardHandshakeMonitor.kt

```diff
@@ -57,8 +57,12 @@ class WireguardHandshakeMonitor @Inject constructor(
 ) : VpnServiceCallbacks {
 
     interface Listener {
-        suspend fun onTunnelFailure(lastHandshakeEpocSeconds: Long)
-        suspend fun onTunnelFailureRecovered()
+        suspend fun onTunnelFailure(
+            coroutineScope: CoroutineScope,
+            lastHandshakeEpocSeconds: Long,
+        )
+
+        suspend fun onTunnelFailureRecovered(coroutineScope: CoroutineScope)
```

Comment:
Undo change, coroutineScope is not necessary as parameter in neither methods

Reply:
Will improve this later on. So far approaches that we’ve tried is blocking the `WireguardHandshakeMonitor` which we do not want since it is the one responsible for calling `onTunnelFailureRecovered` which cancels the continuous recovery attempt.